src/HotChocolate/MongoDb/src/MongoDb/Extensions/MongoFilterOperationFieldExtensions.cs

```diff
@@ -0,0 +1,35 @@
+using HotChocolate.Data.Filters;
+using HotChocolate.Data.Sorting;
+
+namespace HotChocolate.MongoDb
+{
+    internal static class MongoFilterOperationFieldExtensions
+    {
+        public static string GetName(
+            this IFilterField field)
+        {
+            string fieldName = field.Name;
+            if (field.Member is { } p)
+            {
+                fieldName = p.Name;
+            }
+
+            return fieldName;
+        }
+    }
+
+    internal static class MongoSortFieldExtensions
+    {
+        public static string GetName(
+            this ISortField field)
+        {
+            string fieldName = field.Name;
```

Comment:
If the property in the collection has a `BsonElement` attribute, the sort doesn't work.

Reply:
yes this is correct, i think we should annotate this directly as meta information on the field
i abstracted field.Name in advance to add support for this later :) 
Thanks for pointing this out, well spotted

Reply:
As you will see in my PR, I use an IAggregateFluent (you can get it from a collection with just a .Aggregate()) like this when you do a Builders<T>.Sort.Descending(t=> t.X).Ascending(t => t.Y);, the Mongo Driver get automatically the correct field names.